omp/test/factorization/par_ilu_kernels.cpp

```diff
@@ -159,18 +186,20 @@ TEST_F(ParIlu, KernelComputeParILUIsEquivalentToRef)
     auto coo_omp = Coo::create(omp);
     csr_omp->convert_to(gko::lend(coo_omp));
     initialize_lu(&l_ref, &u_ref, &l_omp, &u_omp);
-    auto u_transpose_ref = u_ref->transpose();
-    auto u_transpose_omp = u_omp->transpose();
+    auto u_transpose_lin_op_ref = u_ref->transpose();
+    auto u_transpose_ptr_ref = static_cast<Csr *>(u_transpose_lin_op_ref.get());
+    auto u_transpose_lin_op_omp = u_omp->transpose();
+    auto u_transpose_ptr_omp = static_cast<Csr *>(u_transpose_lin_op_omp.get());
 
     gko::kernels::reference::par_ilu_factorization::compute_l_u_factors(
         ref, iterations, gko::lend(coo_ref), gko::lend(l_ref),
-        gko::lend(u_ref));
+        u_transpose_ptr_ref);
     gko::kernels::omp::par_ilu_factorization::compute_l_u_factors(
         omp, iterations, gko::lend(coo_omp), gko::lend(l_omp),
-        gko::lend(u_omp));
+        u_transpose_ptr_omp);
 
-    GKO_ASSERT_MTX_NEAR(l_ref, l_omp, .3);
-    GKO_ASSERT_MTX_NEAR(u_ref, u_omp, .3);
+    GKO_ASSERT_MTX_NEAR(l_ref, l_omp, 5e-3);
+    GKO_ASSERT_MTX_NEAR(u_transpose_ptr_ref, u_transpose_ptr_omp, 5e-3);
```

Comment:
How is it possible that we now use a stricter threshold? Maybe this only works on your system (with less parallelism)?

Reply:
It also works on the CI.
The important change here is that we now use the transposed `U`, which this kernel requires. Previously, the transposed was calculated, but never used, so the kernel was used incorrectly.

Reply:
I just tried to set the number of iterations to `10` (on my laptop), and now both the reference and the omp implementation produce the __exact__ same result.

Reply:
That is good - it should asymptotically converge to the same factors.